parcel-bundler/parcel
# Consider runtimes as source (isSource)

This:
* Adds the ability to override an asset's `isSource` property through the `AssetRequest` that originates it.
* Overrides `isSource` to be `true` for runtimes so that they are e.g. run through Babel to target the appropriate engines.

Test Plan:
Creating an automated test for this requires Parcel to run in `node_modules` to reproduce `isSource` being set to `false` for runtimes. Instead:
* Make sure the code for Parcel itself is running from a directory including `node_modules` in its path.
* Create a project with a package.json including:
```
  "browsers": [
    "ie 11"
  ],
```
* Bundle an entry file with a dynamic import, which creates a loader runtime for the js.
* Verify that before applying this patch, the js loader in the entry bundle included an arrow function expression.
* Verify that after applying this patch, it does not.
